### ISSUE TITLE
Replace governance auth sigs with proposal NFTs

### ISSUE THREAD

**Issue by plaidfinch** [penumbra-zone/penumbra#1793]
This should follow the design of unbonding tokens, roughly: there should be a unique denom for each proposal. The `ProposalSubmit` action consumes the proposal deposit amount of penumbra and produces a single unit of the denomination `proposal_N_voting` where `N` is the number of the proposal being made (if a proposal is made for the incorrect number, the tx fails). Then, a `ProposalWithdraw` action marks a proposal as withdrawn in the public chain state, consuming a single unit of `proposal_N_voting` and producing a single unit of `proposal_N_withdrawn`. Finally, a `ProposalDepositClaim` action (only valid when a proposal has not been vetoed, and only after voting concludes) consumes a single unit of `proposal_N_voting` or `proposal_N_withdrawn` and produces a single unit of one of `proposal_N_passed`, `proposal_N_failed`, or `proposal_N_vetoed`, as well as the proposal deposit amount for that proposal (we need to store historical proposal deposit amounts in case this changes by governance), provided the proposal was not vetoed. Stateful checks ensure that the input and output match the actual state transition available.

**Comment by conorsch** [comment]
We should spruce up the docs a bit to mention the workflow changes; will file another ticket to track.
